sdk/python/kfp/dsl/for_loop_test.py

```diff
@@ -144,6 +144,24 @@ def test_loop_parameter_argument_from_pipeline_channel(
         self.assertEqual(loop_argument.items_or_pipeline_channel, channel)
         self.assertEqual(str(loop_argument), expected_serialization_value)
 
+    @parameterized.parameters(
+        {
+            'channel':
+                pipeline_channel.PipelineParameterChannel(
+                    name='param1',
+                    channel_type='String',
+                    task_name='task1',
+                ),
+        },)
+    def test_loop_parameter_argument_from_single_pipeline_channel_raises_error(
+            self, channel):
+        with self.assertRaisesRegex(
+                ValueError,
+                r'Cannot iterate over a single Parameter using `dsl\.ParallelFor`\. Expected a list of Parameters as argument to `items`\.'
+        ):
+            loop_argument = for_loop.LoopParameterArgument.from_pipeline_channel(
+                channel)
+
```

Comment:
Can we add a test for this with some example user code to `compiler_test.py`?

Reply:
Done, will add the corresponding test for single artifacts too in the next pr